### PR TITLE
Feature/combined nmap sslyze

### PR DESCRIPTION
**Pull request by dpatanin**
Fixed the comined nmap-ssh scan and added a similar combined nmap-sslyze

### CODE REVIEW

File: scb-scanprocesses/combined-nmap-ssh-process/src/main/java/io/securecodebox/scanprocess/NmapToSshTransformListener.java [hunk 1]
```diff
@@ -53,4 +38,22 @@ public void notify(DelegateExecution delegateExecution) throws Exception {
...
+            String service = (String) finding.getAttributes().get("service");
+            return "ssh".equals(service);
+        })
+        .map(finding -> {
+            String ip_address = (String) finding.getAttributes().get("ip_address");
+            String port = finding.getAttributes().get("port").toString();
+
+            Target target = new Target();
+            target.setName("SSH Scan for " + ip_address);
+            target.setLocation(ip_address + ":" + port);
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-ssh-process/src/main/java/io/securecodebox/scanprocess/NmapToSshTransformListener.java]:
```suggestion
            String ipAddress = (String) finding.getAttributes().get("ip_address");
            String port = finding.getAttributes().get("port").toString();

            Target target = new Target();
            target.setName("SSH Scan for " + ip_address);
            target.setLocation(ipAddress + ":" + port);
```

File: scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+public class NmapToSslyzeTransformListenerTest {
+
+    private static NmapToSslyzeTransformListener listener;
+    private static List<Finding> findings;
+    private static final String TARGET_HOST = "test-host";
+    private static final String TARGET_LOCATION = "test-location";
+
+    @BeforeAll
+    private static void setUp() {
+        listener = new NmapToSslyzeTransformListener();
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java]: Inline var findings in each of the tests

File: scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        findings = new ArrayList<Finding>();
+        findings.add(finding1);
+        findings.add(finding2);
+        findings.add(finding3);
+        findings.add(finding4);
+    }
+
+    @Test
+    protected void filteredServicesShouldBecomeNewTargets() {
+        assertEquals(3, listener.createTargetsFromFindings(findings).size(), "services are detected correctly");
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java]: Would be cool to also have assertion to test that the target create have the correct locations

File: scb-engine/pom.xml [hunk 4]
```diff
@@ -222,6 +222,12 @@
                     <artifactId>combined-nmap-ssh-process</artifactId>
                     <version>0.0.1-SNAPSHOT</version>
                     <scope>runtime</scope>
+                </dependency>
+                 <dependency>
```
Comment (J12934) [inline scb-engine/pom.xml]: Bad indentation
```suggestion
                <dependency>
```

File: scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java (hunk 2 above)
Comment (dpatanin) [reply scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java]: Tests should are now refactored

File: scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java (hunk 3 above)
Comment (dpatanin) [reply scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java]: added more specific tests

File: scb-scanprocesses/combined-nmap-ssh-process/src/main/resources/forms/nmap-ssh/configure-target.html [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            <label>Target Hosts</label>
+            <input
+              required
+              class="form-control"
+              type="text"
+              placeholder="Hostname/IP address"
+              ng-minlength="3"
+              ng-maxlength="100"
+              ng-model="target.location"
+              ng-keydown="checkForEnter($event)"
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-ssh-process/src/main/resources/forms/nmap-ssh/configure-target.html]: This line should also be removed
```suggestion
```

File: scb-scanprocesses/combined-nmap-ssh-process/src/main/resources/forms/nmap-ssh/configure-target.html [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            attributes: {
+                }
+        }];
+
+        $scope.checkForEnter = function ($event) {
+            if ($event.key === 'Enter') {
+                $event.stopPropagation();
+                $event.preventDefault();
+            }
+        };
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-ssh-process/src/main/resources/forms/nmap-ssh/configure-target.html]: This function isn't needed anymore as it was only used to handle the multi target form stuff.
```suggestion
```

File: scb-scanprocesses/combined-nmap-ssh-process/src/main/resources/forms/nmap-ssh/configure-target.html [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+          <div class="col-xs-1">
+            <button
+              class="btn btn-danger btn-lg"
+              ng-click="targetList.splice($index, 1)"
+              ng-disabled="targetList.length === 1"
+              style="position: absolute; right: 15px; top: 0;"
+            >
+              <span class="glyphicon glyphicon-trash"></span>
+            </button>
+          </div>
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-ssh-process/src/main/resources/forms/nmap-ssh/configure-target.html]: This button should also go.
```suggestion
```

File: scb-scanprocesses/combined-nmap-ssh-process/src/test/java/io/securecodebox/scanprocess/NmapToSshTransformListenerTest.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+import io.securecodebox.model.findings.Reference;
+import io.securecodebox.model.findings.Severity;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class NmapToSshTransformListenerTest {
```
Comment (J12934) [inline scb-scanprocesses/combined-nmap-ssh-process/src/test/java/io/securecodebox/scanprocess/NmapToSshTransformListenerTest.java]: Great Test 👍

Comment (dpatanin) [reply scb-scanprocesses/combined-nmap-ssh-process/src/test/java/io/securecodebox/scanprocess/NmapToSshTransformListenerTest.java]: Thanks 🙊

**Pull request merged**